packages/astro/src/runtime/server/hydration.ts

```diff
@@ -9,7 +8,13 @@ import { hydrationSpecifier, serializeListValue } from './util.js';
 
 // Serializes props passed into a component so that they can be reused during hydration.
 // The value is any
-export function serializeProps(value: any) {
+export async function serializeProps(value: any) {
+	// We use a dynamic import here because 'serialize-javascript' generates a
+	// random number in module scope which is forbidden when using cloudflare
+	// workers/pages functions as a deploy target. By importing the dependency
+	// dynamically module scope vars are generated at call time, even when
+	// inlined.
+	const { default: serializeJavaScript } = await import('serialize-javascript');
```

Comment:
`serialize-javascript` was actually removed in https://github.com/withastro/astro/pull/3571, would you mind removing that change and rebasing?

Reply:
Ah amazing, that would have saved me a headache haha